src/ui/camera.js

```diff
@@ -555,7 +561,11 @@ class Camera extends Evented {
         return this._cameraForBoxAndBearing(bounds.getNorthWest(), bounds.getSouthEast(), bearing, options);
     }
 
-    _extendCameraOptions(options?: CameraOptions) {
+    _extendCameraOptions(options?: CameraOptions): {
+        maxZoom: number,
+        offset: PointLike,
+        padding: Required<PaddingOptions>
+    } & CameraOptions {
```

Comment:
Perhaps worth cutting this out into a named type (e.g. `FullCameraOptions`) to make it clearer why it's typed like this and what the method actually returns?